drive.go

```
@@ -48,12 +49,200 @@ func newDrive(projectKey, driveName, rootEndpoint string) *Drive {
+	o, err := d.client.request(&requestInput{
+		Path:        url,
+		QueryParams: queryParams,
+		Method:      "POST",
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var sr startUploadResponse
+	err = json.Unmarshal(o.Body, &sr)
+	if err != nil {
+		return "", err
+	}
+
+	return sr.UploadID, nil
+}
+
+// End a chuncked upload.
+func (d *Drive) finishUpload(name, uploadId string) error {
+	url := fmt.Sprintf("/uploads/%s", uploadId)
+	queryParams := map[string]string{"name": name}
+	_, err := d.client.request(&requestInput{
+		Path:        url,
+		QueryParams: queryParams,
+		Method:      "PATCH",
+	})
+	return err
+}
+
+// Abort a chunked upload.
+func (d *Drive) abortUpload(name, uploadId string) error {
+	url := fmt.Sprintf("/uploads/%s", uploadId)
+	queryParams := map[string]string{"name": name}
+	_, err := d.client.request(&requestInput{
+		Path:        url,
+		QueryParams: queryParams,
+		Method:      "DELETE",
+	})
+	return err
+}
+
+// Uploads a chunked part.
+func (d *Drive) uploadPart(name string, chunk []byte, uploadId string, part int, contentType string) error {
+	url := fmt.Sprintf("/uploads/%s/parts", uploadId)
+	queryParams := map[string]string{"name": name, "part": fmt.Sprintf("%d", part)}
+	_, err := d.client.request(&requestInput{
+		Path:        url,
+		QueryParams: queryParams,
+		Method:      "POST",
+		RawBody:     chunk,
+		ContentType: contentType,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PutInput input for Put operation.
+type PutInput struct {
+	// name of file
+	Name string
+	// io.Reader with file content
+	Body io.Reader
+	// content type of file
+	ContentType string
+}
+
+// Put a file in the Drive.
+//
+// Returns the name of file that was put in the drive.
+func (d *Drive) Put(i *PutInput) (string, error) {
+	if i.Name == "" {
+		return i.Name, ErrEmptyName
+	}
+
+	if i.Body == nil {
+		return i.Name, ErrEmptyData
+	}
+
+	// start upload
+	uploadId, _ := d.startUpload(i.Name)
```

we should not ignore the error